ws/src/server_builder.rs

```diff
@@ -37,6 +38,28 @@ impl From<io::Error> for Error {
 	}
 }
 
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            Error::Io(ref e) => e.fmt(f),
+            Error::WebSocket(ref e) => e.fmt(f),
+        }
+    }
+}
+
+impl ::std::error::Error for Error {
+    fn description(&self) -> &str {
+        "Starting the JSON-RPC WebSocket server failed"
```

Comment:
I wanted them to work the same as the http errors. Also a chain of errors (this is one error and it contains a cause that is another error) should not just chain down transparently. The intermediate steps in the chain must have relevant information, otherwise there is no information why one got the bottom level error one got.

Instead of just delegating this down one might even want to change the display implementation to be `"Starting the JSON-RPC WebSocket server failed: {}"` and then format in the cause.